src/python/pants/backend/awslambda/python/awslambda_python_rules.py

```diff
@@ -51,7 +52,9 @@ async def create_python_awslambda(
     field_set: PythonAwsLambdaFieldSet, lambdex_setup: LambdexSetup
 ) -> CreatedAWSLambda:
     # Lambdas typically use the .zip suffix, so we use that instead of .pex.
-    pex_filename = f"{field_set.address.target_name}.zip"
+    pex_filename = os.path.join(
```

Comment:
should we be using pathlib here?
@Eric-Arellano

Reply:
Not a big deal. Generally, yeah, but `os.path.join` is faster than `PurePath`. 

I've only been using `PurePath` when we need more complex things like `path.parent.parent`, or path.read_text()`.